src/components/OfferRoute/OfferRoute.tsx

```diff
@@ -0,0 +1,128 @@
+import React, { useEffect, useState } from 'react';
+import { AccountTypes, ExchangeOffer, TokenListToken } from 'etherspot';
+import styled, { useTheme } from 'styled-components';
+import { ethers } from 'ethers';
+
+import { useEtherspot } from '../../hooks';
+
+import { addressesEqual } from '../../utils/validation';
+import { formatAmountDisplay } from '../../utils/common';
+import { IAssetWithBalance } from '../../providers/EtherspotContextProvider';
+import { Chain } from '../../utils/chain';
+
+import { SelectOption } from '../SelectInput/SelectInput';
+import { RoundedImage } from '../Image';
+import { Text } from '../Text';
+import { Theme } from '../../utils/theme';
+
+type RouteOptionProps = {
+  option: SelectOption;
+  availableOffers: ExchangeOffer[] | null;
+  targetAssetPriceUsd: number | null;
+  selectedFromAsset: IAssetWithBalance | null;
+  selectedAccountType: string;
+  selectedToAsset?: TokenListToken | null;
+  selectedNetwork?: Chain | null;
+  availableToAssets?: TokenListToken[] | null;
+};
+
+export const OfferRoute = (props: RouteOptionProps) => {
+  const {
+    option,
+    availableOffers,
+    availableToAssets,
+    selectedToAsset,
+    targetAssetPriceUsd,
+    selectedNetwork,
+    selectedFromAsset,
+    selectedAccountType,
+  } = props;
+  const [gasPrice, setGasPrice] = useState('-');
+  const [isEstimating, setIsEstimating] = useState(true);
+  const { getSdkForChainId } = useEtherspot();
+  const theme: Theme = useTheme();
+
+  const getGasSwapUsdValue = async (offer: ExchangeOffer) => {
+    const sdkByChain = getSdkForChainId(selectedNetwork?.chainId ?? 1);
+    setIsEstimating(true);
+
+    if (sdkByChain && selectedFromAsset && selectedAccountType === AccountTypes.Contract) {
+      await sdkByChain.computeContractAccount();
+
+      await sdkByChain.clearGatewayBatch();
+
+      for (let i = 0; i < offer.transactions.length; i++) {
+        await sdkByChain.batchExecuteAccountTransaction(offer.transactions[i]);
+      }
```

Comment:
using for will block UI, use async method
```suggestion
      await Promise.all((offer.transactions).map((transaction) => sdkByChain.batchExecuteAccountTransaction(transaction)));
      }
```